checks.d/haproxy.py

```diff
@@ -319,51 +336,37 @@ def _process_status_metric(self, hosts_statuses, collect_status_metrics_by_host,
                 service, status = host_status
             status = status.lower()
 
-            tags = []
-            if count_status_by_service:
-                tags.append('service:%s' % service)
-
             if self._is_service_excl_filtered(service, services_incl_filter, services_excl_filter):
                 continue
 
+            tags = []
+            if count_status_by_service:
+                tags.append('service:%s' % service)
             if collect_status_metrics_by_host:
                 tags.append('backend:%s' % hostname)
```

Comment:
```python
            try:
                service, hostname, status = host_status
            except Exception:
                service, status = host_status

```
Are we sure that the `hostname` variable is always defined when `collect_status_metrics_by_host ` ?